src/plugins/workspace/public/components/workspace_detail/select_data_source_panel.tsx

```diff
@@ -42,11 +45,64 @@ export const SelectDataSourceDetailPanel = ({
   currentWorkspace,
 }: SelectDataSourcePanelProps) => {
   const {
-    services: { notifications, workspaceClient },
+    services: { notifications, workspaceClient, http },
   } = useOpenSearchDashboards<{ CoreStart: CoreStart; workspaceClient: WorkspaceClient }>();
   const { formData, setSelectedDataSources } = useWorkspaceFormContext();
   const [isLoading, setIsLoading] = useState(false);
   const [isVisible, setIsVisible] = useState(false);
+  const [dataSourceConnections, setDataSourceConnections] = useState<DataSourceConnection[]>([]);
+  const [toggleIdSelected, setToggleIdSelected] = useState('all');
+
+  const fetchDQC = useCallback(async () => {
+    setIsLoading(true);
+    try {
+      const directQueryConnectionsPromises = assignedDataSources.map((ds) =>
+        getDirectQueryConnections(ds.id, http!)
+      );
+      const directQueryConnectionsResult = await Promise.all(directQueryConnectionsPromises);
```

Comment:
Nit: The API will fire many requests at the same time if there is a large number of data source. Maybe we could optimize it in the future version.